src/core/inductive.ml

```diff
@@ -55,17 +56,74 @@ let gen_ind_typ_codom : popt -> sym -> (tbox list -> tbox) -> string -> tbox =
     | _ -> fatal pos "The type of %a is not supported" pp_symbol ind_sym
   in aux [] !(ind_sym.sym_type)
 
+module T =
+  struct
+
+    (** [fold_app a [b1;...;bn]] returns (... ((a b1) b2) ...) bn. *)
+    let fold_app  : tbox -> tbox list -> tbox = List.fold_left _Appl
+
+    (** [fold_app_special f ts] returns the same result that
+        fold_app (_Symb [f]) [ts]. *)
+    let fold_app_special : sym -> tbox list -> tbox = fun f ts ->
```

Comment:
fold_app_special -> _Appl_symb ? can be defined in terms.ml too

Reply:
Done (I haven't better idea for the name)